content/Voice Over - Fernanda Toledo.md

```diff
@@ -0,0 +1,63 @@
+
+### What is Voice Over?
+
+For users that have a very low or no sight, there's Voice Over, a gesture-based screen reader that allows users to know what's on the screen by playing audible descriptions. The user can navigate through the screen by swiping and they can trigger different actions depending on a given [set of gestures](https://support.apple.com/guide/iphone/learn-voiceover-gestures-iph3e2e2281/13.0/ios/13.0).
+
+### Try it on one of your apps!
+
+Open one of the apps you've developed and turn on Voice Over, read the entire content of the screen by swiping up using two fingers and voila, by default Voice Over will read some things, so that's a good head start, some buttons can be tapped by using double tap and some labels may be read correctly, but I can guarantee you that if accessibility was not taken into account in the design and development stages, then there will be issues. There might be things that are not being read in the correct order, things that are not being read at all, and some elements might not be grouped correctly because of a wrong view hierarchy. Some common mistakes can be that an uppercase text is by default read one letter at a time (OMG, for instance, should be read Oh my god, but instead the 3 letters will be read), and in case you have checkboxes probably the state of the checkbox is not being announced. If your app has those issues, users with disabilities won't be able to use it, because those things make it impossible for them to move forward.
+
+### Get your hands dirty
+
+One fun little experiment we did at Rootstrap was developing a very silly app. The purpose of it was to create to-do lists, users could add items that consisted of just a checkbox and a text and they could mark them as either complete or incomplete by checking on the checkbox, classic to-do stuff. We handed a blindfold to our CTO and asked him to add one item and mark it as complete, it was painful to watch, he struggled for a while to figure out how to open the keyboard. Then, we improved the default implementation a little:
+
+- Gave our UI an intuitive hierarchy structure: Group UI elements that make sense together to provide a faster and better navigation
+
+- isAccessibleElement: Make non relevant UI elements not focusable by Voice Over, like shadows and images, anything that's decorative
+
+- accessibilityLabel: Provide a useful description to Voice Over to read when the item is focused
+
+- accessibilityHint: Provide additional context or actions for the selected element
+
+- accessibilityTraits: Tells Voice Over how the elements behave or should be treated, this can be button, isSelected, text, etc
+
+- UIAccessibilityLayoutChangedNotification: Notifies Voice Over about changes in layout, so that the last item added to the list gets focused
```

Comment:
```suggestion
- UIAccessibilityLayoutChangedNotification: Notifies VoiceOver about changes in layout, so that the last item added to the list gets focused
```